app/src/main/java/com/aragh/kotlin2/di/Modules.kt

```diff
@@ -1,23 +1,43 @@
 package com.aragh.kotlin2.di
 
+import com.aragh.kotlin2.interactor.Albums
+import com.aragh.kotlin2.interactor.UserAlbums
+import com.aragh.kotlin2.interactor.Users
 import com.aragh.kotlin2.api.AlbumsApi
-import com.aragh.kotlin2.api.AlbumsRepo
+import com.aragh.kotlin2.api.UserAlbumsApi
 import com.aragh.kotlin2.api.UsersApi
-import com.aragh.kotlin2.api.UsersRepo
+import com.aragh.kotlin2.screen.albumdetails.AlbumDetailsPresenter
+import com.aragh.kotlin2.screen.albumdetails.PresenterContract
+import com.aragh.kotlin2.screen.useralbums.UserAlbumsPresenter
+import com.aragh.kotlin2.screen.users.UsersPresenter
 import org.koin.dsl.module.applicationContext
 import retrofit2.Retrofit
+import retrofit2.adapter.java8.Java8CallAdapterFactory
 import retrofit2.converter.gson.GsonConverterFactory
 
 
 val usersModule = applicationContext {
-  factory {
+  bean {
     Retrofit.Builder()
         .baseUrl("https://jsonplaceholder.typicode.com")
         .addConverterFactory(GsonConverterFactory.create())
+        .addCallAdapterFactory(Java8CallAdapterFactory.create())
         .build()
   }
-  factory { get<Retrofit>().create(UsersApi::class.java) }
-  factory { UsersRepo(get()) }
-  factory { get<Retrofit>().create(AlbumsApi::class.java) }
-  factory { AlbumsRepo(get()) }
+  bean { get<Retrofit>().create(UsersApi::class.java) }
+  bean { Users(get()) }
+  bean { get<Retrofit>().create(UserAlbumsApi::class.java) }
+  bean { UserAlbums(get()) }
+  bean { get<Retrofit>().create(AlbumsApi::class.java) }
+  bean { Albums(get()) }
+
+  context("albumDetails") {
+    bean<PresenterContract> { AlbumDetailsPresenter(get()) }
+  }
+  context("userAlbums") {
+    bean<com.aragh.kotlin2.screen.useralbums.PresenterContract> { UserAlbumsPresenter(get()) }
```

Comment:
I don't like that on line 35 albumdetails.PresenterContract is treated in a better way than this useralbums.PresenterContract :D Maybe you could use import alias, `import com.aragh.kotlin2.screen.albumdetails.PresenterContract as AlbumDetailsPresenterContract` and `com.aragh.kotlin2.screen.useralbums.PresenterContract as UserAlbumsPresenterContract`? Or rename them, but I guess you don't want to do that.

You can also see my comment in 
https://github.com/karolwartak/kotlin-android-testapp/pull/1/files#diff-2674322db6a6791a2feb032f881c9aa3